synapse/config/database.py

```diff
@@ -163,12 +164,12 @@ def read_config(self, config, **kwargs):
             self.databases = [DatabaseConnectionConfig("master", database_config)]
             self.set_databasepath(database_path)
 
-    def generate_config_section(self, data_dir_path, **kwargs):
+    def generate_config_section(self, data_dir_path, **kwargs) -> str:
         return DEFAULT_CONFIG % {
             "database_path": os.path.join(data_dir_path, "homeserver.db")
         }
 
-    def read_arguments(self, args):
+    def read_arguments(self, args: argparse.Namespace) -> None:
```

Comment:
Aside: I wish there was a way for mypy to know which attributes would be available on `args` at typechecktime.

Reply:
You mean if you could note what type of arguments exist in the `Namespace`?

Reply:
I was thinking more that I'd like mypy to be able to deduce which args would exist, but an explict annotation would be good too!